src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/show_request.cc

```diff
@@ -94,6 +99,125 @@ void ShowRequest::HandleMetadata(const char* field_id,
   }
 }
 
+bool ShowRequest::CheckTemplateConfigurationForApp(
+    application_manager::Application& app) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  if ((*message_)[strings::msg_params].keyExists(strings::window_id)) {
+    current_window_id_ =
+        (*message_)[strings::msg_params][strings::window_id].asInt();
+  }
+
+  const auto set_window_layout = [&app, this]() -> bool {
+    const auto new_template_layout =
+        template_config_[strings::template_layout].asString();
+    const auto old_template_layout = app.window_layout(current_window_id_);
+    LOG4CXX_DEBUG(logger_, "New layout: " << new_template_layout);
+    LOG4CXX_DEBUG(logger_, "Old layout: " << old_template_layout);
+
+    const bool layouts_equal = (new_template_layout == old_template_layout);
+
+    if (!new_template_layout.empty() && !layouts_equal) {
+      // Template switched, hence allow any color change
+      LOG4CXX_DEBUG(logger_,
+                    "Show Request: Setting new Layout: " << new_template_layout
+                                                         << " for window ID: "
+                                                         << current_window_id_);
+      layout_change_required_ = true;
+      return true;
+    }
+    LOG4CXX_DEBUG(logger_, "Show Request: No Layout Change");
+    return false;
+  };
+
+  const auto set_day_color_scheme = [&app, this]() -> bool {
+    if (!template_config_.keyExists(strings::day_color_scheme)) {
+      return false;
+    }
+    if (app.day_color_scheme(current_window_id_).getType() !=
+            smart_objects::SmartType_Null &&
+        template_config_[strings::day_color_scheme] !=
+            app.day_color_scheme(current_window_id_)) {
+      // Color scheme param exists and has been previously set,
+      // hence do not allow color change
+      LOG4CXX_DEBUG(logger_, "Day Color Scheme change is rejected");
```

Comment:
Also this information should probably be passed back to the app requesting the color change that was rejected

Reply:
Nevermind i see the error response below